### ISSUE TITLE
Running docker in systemd-nspawn

### ISSUE THREAD

**Issue by kiesstein** [dockur/windows#220]
I was trying to start the container in a systemd-nspawn container where /dev/kvm is bind mounted inside. 
Inside the jail i run kvm-ok:
```shell
root@debian:/mnt/safe/docker/compose/windows$ kvm-ok 
INFO: /dev/kvm exists
KVM acceleration can be used
``` 
The Jail is the docker host with the following config:
```dockerfile
version: "3"
services:
  windows:
    image: dockurr/windows
    container_name: windows
    devices:
      - /dev/kvm
    cap_add:
      - NET_ADMIN
    ports:
      - 8006:8006
      - 3389:3389/tcp
      - 3389:3389/udp
    stop_grace_period: 2m
    restart: on-failure
    volumes:
      - /mnt/tank/all/kvm/win:/storage

``` 
But when i start the container the VM just shuts down:

```shell
1933312K ........ ........ ........ ........ 41% 65.1M 37s

1966080K ........ ........ ........ ........ 42% 69.9M 36s

1998848K ........ ........ ........ ........ 42% 33.9M 36s


...



4718592K ........ ........ ........ ........ 99% 69.3M 0s

4751360K .....                              100% 38.2M=62s

❯ Extracting Windows 11 bootdisk...

❯ Extracting Windows 11 environment...

❯ Extracting Windows 11 setup...

❯ Extracting Windows 11 image...

❯ Adding XML file for automatic installation...

❯ Building Windows 11 image...

❯ Creating a 64G growable disk image in raw format...

❯ Booting Windows using QEMU emulator version 8.2.1 ...

3h3h3hBdsDxe: failed to load Boot0002 "UEFI QEMU QEMU HARDDISK " from PciRoot(0x0)/Pci(0xA,0x0)/Scsi(0x0,0x0): Not Found

BdsDxe: loading Boot0001 "UEFI QEMU QEMU CD-ROM " from PciRoot(0x0)/Pci(0x5,0x0)/Scsi(0x0,0x0)

BdsDxe: starting Boot0001 "UEFI QEMU QEMU CD-ROM " from PciRoot(0x0)/Pci(0x5,0x0)/Scsi(0x0,0x0)

❯ Shutdown completed!

``` 

How can i get more logs or what do i miss? Is it not possible to run with bind mounted /dev/kvm or nspawn-jail?
I also installed qemu-kvm inside the jail:
```shell
sudo apt -y install qemu-kvm libvirt-daemon  bridge-utils virtinst libvirt-daemon-system
``` 
System: TrueNAS-Scale; Ryzen 1600x, SVM enabled
JailOS: Debian Bookworm

**Comment by kiesstein** [comment]
So it did some digging and found out that the the container also does need the device `/dev/vhost-net`.
I found this while looking at a different docker-kvm project in the troubleshoot section:
[https://github.com/BBVA/kvm?tab=readme-ov-file#notes--troubleshooting](url)

So what i did was running bind mounting  `/dev/vhost-net` into the nspawn-container and then qemu was able to boot up.

The weird thing is when i enable the console for more debug like @kroese mentioned:
```dockerfile
environment:
  CONSOLE: "Y"
``` 
The VM would not start and only the following logs appeared when `CONSOLE: "Y"`  :
```shell
❯ Starting Windows for Docker v2.05...

❯ For support visit https://github.com/dockur/windows

❯ Booting Windows using QEMU emulator version 8.2.1 ...

char device redirected to /dev/pts/0 (label serial0)
```
... and when i remove it, it works but the logs are kind of empty:
```shell
No log line matching the '' filter
``` 

I guess there is some kind of bug with that environment variable.
I don't know if this issue is only on my part with my special setup or this should be a separate issue.

**Comment by kiesstein** [comment]
You are right @kroese - it could not use `mknot` to create the device because the nspawn container needs the rights to do so. 
I removed the bind mount of the device and gave it the rights to `rwm` of `/dev/vhost-net` like in: [systemd-nspawn container](https://unix.stackexchange.com/questions/304252/access-usb-device-from-systemd-nspawn-container) with `'--property=DeviceAllow=/dev/vhost-net rwm'` and the windows VM boots up without issues!

**Comment by kiesstein** [comment]
Ok so I tried to reproduce the problem with `2.05` and tested with `winxp` and I could not reproduce it.

I tried to install `win11`  again with the `--property='DeviceAllow=/dev/vhost-net rwm'` but still does not work - mhm, maybe it is only an installation issue or previously I only tried with `winxp` because it is faster to test - sadly I don't remember anymore. 
So it is at least a `win11` issue.
The statement that it works with `--property='DeviceAllow=/dev/vhost-net rwm'` and `win11` is false!

Next test was instead `--property='DeviceAllow=/dev/vhost-net rwm'`, bind mounting `--bind=/dev/vhost-net`, but sadly same issue.

Next was `--capability=all` without  `--property='DeviceAllow=/dev/vhost-net rwm'` or `--bind=/dev/vhost-net` -> not working.

Next was `--capability=all`, `--property='DeviceAllow=/dev/vhost-net rwm'` and  `--bind=/dev/vhost-net` -> not working.
I also did `modprobe vhost_vsock` on the jail-host because I remember doing that, but no.

Then I tested all bind mounts listed in [this](https://github.com/Jip-Hop/jailmaker/blob/main/templates/incus/config) + `vhost-net`:
```shell 
--bind=/dev/kvm --capability=all --bind=/dev/vhost-net --bind=/dev/fuse --bind=/dev/vsock --bind=/dev/vhost-vsock

``` 
-> not working.
I tried with `winxp` with same settings to make sure this at least still works, and it did.

Then I also tried to bind vhost-net in the dockerfile with `win11` again like:
```dockerfile
devices:
  - /dev/kvm
  - /dev/vhost-net
``` 

And still does not work. So I am not sure anymore what all I did in my testing but I can't get it running anymore with `2.05` with `win11`, and I do not remember exactly if it ever run with `with11` - so there is that.

Then I tired with all the previous mentioned settings `win10` and it did not work. 
With `win8` it did work! So something beginning with `win10` is the issue. 

I am sorry that I tested wrong (changing more than one setting(winxp)).

Then i did a `docker compose pull` and restarting the container with `win8`
First boot did not work after recreation - but then i restarted the container again and it did boot. I don't know, maybe a fluke? 
```
❯ Starting Windows for Docker v2.07...

❯ For support visit https://github.com/dockur/windows

❯ Booting Windows using QEMU emulator version 8.2.1 ...

3h3h3hBdsDxe: loading Boot0003 "Windows Boot Manager" from HD(1,GPT,3C9F9213-A787-474C-8772-B5A6D80C1E6B,0x800,0x40000)/\EFI\Microsoft\Boot\bootmgfw.efi

BdsDxe: starting Boot0003 "Windows Boot Manager" from HD(1,GPT,3C9F9213-A787-474C-8772-B5A6D80C1E6B,0x800,0x40000)/\EFI\Microsoft\Boot\bootmgfw.efi

❯ Shutdown completed!

❯ Starting Windows for Docker v2.07...

❯ For support visit https://github.com/dockur/windows

❯ Booting Windows using QEMU emulator version 8.2.1 ...

3h3h3hBdsDxe: loading Boot0003 "Windows Boot Manager" from HD(1,GPT,3C9F9213-A787-474C-8772-B5A6D80C1E6B,0x800,0x40000)/\EFI\Microsoft\Boot\bootmgfw.efi

BdsDxe: starting Boot0003 "Windows Boot Manager" from HD(1,GPT,3C9F9213-A787-474C-8772-B5A6D80C1E6B,0x800,0x40000)/\EFI\Microsoft\Boot\bootmgfw.efi

``` 

Then i tried `win10` again with no bind-mount and so on, but is not working with `v2.07`.  It also does not work with all the bind mounts and `privileged: true` and `- /dev/vhost-net` in docker compose file.

In summary all the settings did nothing. Only thing i found out is that the it only **works** with `winxp` and `win8` (`win7`, `vista`, ... I did not test) and it does **not** work with `win10` and `win11`
So I am fairly certain that the devices or bind mounts are not the problem - so sadly we are back to square one.
